src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java

```diff
@@ -176,25 +205,129 @@ public Action createAction(final Action action) {
+      if (enrolledAssociations.isEmpty()) {
+        log.error(NO_RESPONDENTS_FOR_SURVEY);
+        throw new IllegalStateException(NO_RESPONDENTS_FOR_SURVEY);
+      }
+
+      enrolledAssociations.forEach(
+          association ->
+              createAction(
+                  actionCase, actionRule, actionType, UUID.fromString(association.getPartyId())));
+
+    } else {
+      createAction(actionCase, actionRule, actionType, actionCase.getPartyId());
+    }
+  }
+
+  private List<Association> respondentsEnrolledOnCase(ActionCase actionCase) {
+    PartyDTO businessParty =
+        partySvcClientService.getParty(actionCase.getSampleUnitType(), actionCase.getPartyId());
+    CollectionExerciseDTO collectionExercise =
+        collectionExerciseClientService.getCollectionExercise(actionCase.getCollectionExerciseId());
+    return associationsEnrolledForSurvey(businessParty, collectionExercise.getSurveyId());
+  }
+
+  private List<Association> associationsEnrolledForSurvey(PartyDTO party, String surveyId) {
+    return party
+        .getAssociations()
+        .stream()
+        .filter(association -> isAssociationEnabledForSurvey(association, surveyId))
+        .collect(Collectors.toList());
+  }
 
-    actionRepo
-        .findPotentialActionsActiveDate(actionPlan.getActionPlanPK(), currentTime)
+  private boolean isAssociationEnabledForSurvey(Association association, String surveyId) {
+    return association
+        .getEnrolments()
         .stream()
-        .map(pa -> Action.fromPotentialAction(pa, currentTime))
-        .forEach(a -> actionRepo.save(a));
-    actionRepo.flush();
+        .anyMatch(
+            enrolment ->
+                enrolment.getSurveyId().equals(surveyId)
+                    && enrolment.getEnrolmentStatus().equalsIgnoreCase(ENABLED));
+  }
+
+  private void createAction(
+      ActionCase actionCase, ActionRule actionRule, ActionType actionType, UUID partyId) {
+    if (actionRepo.existsByCaseIdAndActionRuleFKAndPartyId(
+        actionCase.getId(), actionRule.getActionRulePK(), partyId)) {
+      log.with("case_id", actionCase.getId().toString())
+          .with("action_rule_id", actionRule.getActionRulePK().toString())
+          .with("party_id", partyId.toString())
+          .debug("Action already exists");
+      return;
+    }
+
+    Action newAction = new Action();
+    newAction.setId(UUID.randomUUID());
+    newAction.setCreatedBy(SYSTEM);
+    newAction.setManuallyCreated(false);
+    newAction.setState(ActionState.SUBMITTED);
+    newAction.setCreatedDateTime(new Timestamp((new Date()).getTime()));
+
+    newAction.setCaseFK(actionCase.getCasePK());
+    newAction.setCaseId(actionCase.getId());
+
+    newAction.setActionPlanFK(actionRule.getActionPlanFK());
+    newAction.setActionRuleFK(actionRule.getActionRulePK());
+    newAction.setActionType(actionType);
+    newAction.setPriority(actionRule.getPriority());
+
+    newAction.setPartyId(partyId);
+
+    actionRepo.saveAndFlush(newAction);
```

Comment:
Is it sensible to have an info level log here? I know it's a happy path but would be very useful when debugging this complex process of creating actions. The alternative would be to go into the prod database and query the actions table to see if an action has been created (not everyone has this access).

If not info there should at least be a debug log line for the successful action creation.

Reply:
debug not info